### PR TITLE
Feature/redesign refactoring list

### PR DESCRIPTION
**Pull request by aborken**
In this change, we:
- renamed components
- added CapsToTitleCase pipe
- using ng-template for our list.component
- added column types:
-- Text: show property as text
-- Long Text: trim property string after some characters and show "Show more"
-- Link: show link
-- Icon: show property as icon
-- Action Button: show an action button in table or card
-- Text and more: show property as text and a second information with a specialized separator
-- Link and more: show property as link and a second information with a specialized separator

### CODE REVIEW

File: nexuse2e-webapp/src/main/frontend/src/app/pipes/caps-to-title-case.pipe.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+})
+export class CapsToTitleCasePipe implements PipeTransform {
+  transform(value: unknown): unknown {
+    if (typeof value === "string") {
+      const titleCasePipe = new TitleCasePipe();
+      const uppercase = value.toUpperCase();
+      if (value === uppercase) {
+        return titleCasePipe.transform(value) as string;
+      } else {
+        return value as string;
```
Comment (StefanSchoof) [inline nexuse2e-webapp/src/main/frontend/src/app/pipes/caps-to-title-case.pipe.ts:16]: I think the `as` is not needed. typescript knows that value is here of the type string
```suggestion
      if (value === uppercase) {
        return titleCasePipe.transform(value);
      } else {
        return value;
```

File: nexuse2e-webapp/src/main/frontend/src/app/pipes/caps-to-title-case.pipe.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+      }
+    } else {
+      return value;
```
Comment (StefanSchoof) [inline nexuse2e-webapp/src/main/frontend/src/app/pipes/caps-to-title-case.pipe.ts:19]: The angular own Pipes are throw if the argument is a not supported, see
https://github.com/angular/angular/blob/12.0.1/packages/common/src/pipes/case_conversion_pipes.ts#L11-L42
Then we can provide a return type. And I think the null handling is also nice done there

File: nexuse2e-webapp/src/main/frontend/src/app/pipes/caps-to-title-case.pipe.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  name: "capsToTitleCase",
+  pure: true,
+})
+export class CapsToTitleCasePipe implements PipeTransform {
+  transform(value: unknown): unknown {
+    if (typeof value === "string") {
+      const titleCasePipe = new TitleCasePipe();
+      const uppercase = value.toUpperCase();
+      if (value === uppercase) {
+        return titleCasePipe.transform(value) as string;
```
Comment (StefanSchoof) [inline nexuse2e-webapp/src/main/frontend/src/app/pipes/caps-to-title-case.pipe.ts:14]: We need the TitleCasePipe only if the string is uppercase, so we can initialize it only in this case.
```suggestion
      const uppercase = value.toUpperCase();
      if (value === uppercase) {
        const titleCasePipe = new TitleCasePipe();
        return titleCasePipe.transform(value) as string;
```

**Review by StefanSchoof — APPROVED** [review]
I like the ng-template approch

File: nexuse2e-webapp/src/main/frontend/src/app/pipes/caps-to-title-case.pipe.ts (hunk 2 above)
Comment (aborken) [reply nexuse2e-webapp/src/main/frontend/src/app/pipes/caps-to-title-case.pipe.ts:19]: Our pipe should transform string values, but other value types should be returned as they are.

File: nexuse2e-webapp/src/main/frontend/src/app/pipes/caps-to-title-case.pipe.ts (hunk 3 above)
Comment (aborken) [reply nexuse2e-webapp/src/main/frontend/src/app/pipes/caps-to-title-case.pipe.ts]: Done.

**Pull request merged**